### PR TITLE
fix: handle product/category page in LinkComponent (#15560)

### LINKED ISSUE #15560: Refactor BannerComponent to support content/product/category page
SmartEdit adds new properties in CMS `SimpleBannerComponent`. Except `urlLink`, now customer can add `content page` (uid), or `product/category page` (code). So, we need to refactor our `BannerComponent` in Spartacus to support the changes as well.

### PR DESCRIPTION
**Pull request by BenjaminEmiliani**
Closes #15560

### CODE REVIEW

File: projects/storefrontlib/cms-components/content/link/link.component.ts [hunk 1]
```diff
@@ -25,4 +33,15 @@ export class LinkComponent {
   getTarget(data: CmsLinkComponent): string | null {
     return data.target === 'true' || data.target === true ? '_blank' : null;
   }
+
+  /**
+   * Set routerlink used by generic link if url from CMS data is undefined
+   * @param linkData CMSLinkComponent data
+   */
+  setRouterLink(linkData: CmsLinkComponent): void {
```
Comment (RadhepS) [inline projects/storefrontlib/cms-components/content/link/link.component.ts:41]: We could maybe call this function from the template directly, what do you think?

Comment (BenjaminEmiliani) [reply projects/storefrontlib/cms-components/content/link/link.component.ts:41]: Ah, yes do you mean (~~setRouterLink~~ -> getRouterLink): 
[url]="data.url ?? getRouterLink(data)"

Comment (RadhepS) [reply projects/storefrontlib/cms-components/content/link/link.component.ts:41]: Yeah exactly

File: projects/core/src/cms/facade/cms.service.ts [hunk 2]
```diff
@@ -278,4 +280,33 @@ export class CmsService {
   setPageFailIndex(pageContext: PageContext, value: string): void {
     this.store.dispatch(new CmsActions.CmsSetPageFailIndex(pageContext, value));
   }
+
+  /**
+   * Get routerLink from CMS data
+   * @param data CMSComponent data
+   */
+  getRouterLink(data: any): Observable<any> {
```
Comment (WeizhengSap) [inline projects/core/src/cms/facade/cms.service.ts:288]: better not add this function here. This service is for `CMS`, but not for some specific cms component.

**Comment by BenjaminEmiliani** [comment]
Closing, ticket was revised by author and best is to restart PR from a clean branch off develop.

**Pull request closed without merging**